### PR TITLE
[lldb] Implement TypeSystemSwiftTypeRef::ShouldPrintAsOneLiner

### PR DESCRIPTION
**Pull request by kastiglione**
Implement `TypeSystemSwiftTypeRef::ShouldPrintAsOneLiner` to match the `SwiftASTContext` implementation.

rdar://68171384

### CODE REVIEW

File: lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.cpp [hunk 1]
```diff
@@ -2327,12 +2327,29 @@ CompilerType
...
+        return eLazyBoolNo;
+      if ((valobj->GetLanguageFlags() & LanguageFlags::eIsIndirectEnumCase) ==
+          LanguageFlags::eIsIndirectEnumCase)
+        return eLazyBoolNo;
+    }
+
+    return eLazyBoolCalculate;
+  };
+  VALIDATE_AND_RETURN(impl, ShouldPrintAsOneLiner, type,
+                      (ReconstructType(type), valobj));
```
Comment (adrian-prantl) [inline lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.cpp:2350]: This looks sufficiently generic that you could hoist it all the way into the TypeSystemSwift base class and delete the SwiftASTContext implementation.

Comment (adrian-prantl) [reply lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.cpp:2350]: There's nothing TypeSystemSwiftTypeRef-specific.

Comment (adrian-prantl) [reply lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.cpp:2350]: That said, I'm also fine with landing it as is and doing that as an NFC refactor later.

**Review by adrian-prantl — APPROVED** [review]

**Comment by kastiglione** [comment]
@swift-ci test

**Comment by kastiglione** [comment]
I'll do an NFC refactor later. There may be a few functions like this.

**Pull request merged**